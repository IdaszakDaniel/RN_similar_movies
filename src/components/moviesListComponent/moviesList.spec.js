/* eslint-disable @typescript-eslint/camelcase */
import React from 'react'
import { render, fireEvent } from 'react-native-testing-library'
import MoviesList from './moviesList'
import { toContainElement } from '@testing-library/jest-native'

expect.extend({ toContainElement })

const movies = [
  { data: { poster_path: 'poster1.png', id: 1 }, title: 'testTitle1' },
  { data: { poster_path: 'poster2.png', id: 2 }, title: 'testTitle2' }
]

const movie = [
  { data: { poster_path: 'poster1.png', id: 1 }, title: 'testTitle1' }
]

const props = {
  moviesList: movies,
  selectItem: () => {}
}

describe('MoviesList:', () => {
  describe('Flatlist element', () => {
    it('renders correctly', () => {
      const tree = render(<MoviesList {...props} />).toJSON()
      expect(tree).toMatchSnapshot()
    })

    it('contains item with text', () => {
      const { queryByTestId, getByText } = render(<MoviesList {...props} />)

      const FlatList = queryByTestId('flatlist')
      const ListItem = getByText('testTitle1')

      expect(FlatList).toContainElement(ListItem)
    })

    it('returns proper data after click', () => {
      const onSelectMock = jest.fn()
      const props = {
        moviesList: movie,
        selectItem: onSelectMock
      }
      const { getByTestId } = render(<MoviesList {...props} />)

      fireEvent(getByTestId('checkBox'), 'onChange', true)
      expect(onSelectMock).toHaveBeenCalled()
      expect(onSelectMock).toHaveBeenCalledWith(movie[0], true)
    })
  })
})
