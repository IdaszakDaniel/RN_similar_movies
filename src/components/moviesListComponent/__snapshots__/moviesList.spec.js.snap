// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesList: Flatlist element renders correctly 1`] = `
<View
  style={
    Object {
      "height": "100%",
      "width": "100%",
    }
  }
>
  <RCTScrollView
    data={
      Array [
        Object {
          "data": Object {
            "id": 1,
            "poster_path": "poster1.png",
          },
          "title": "testTitle1",
        },
        Object {
          "data": Object {
            "id": 2,
            "poster_path": "poster2.png",
          },
          "title": "testTitle2",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    showsVerticalScrollIndicator={false}
    stickyHeaderIndices={Array []}
    testID="flatlist"
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "borderBottomColor": "gray",
              "borderBottomWidth": 1,
              "flexDirection": "row",
              "height": 50,
              "justifyContent": "space-between",
              "marginVertical": 5,
            }
          }
        >
          <View
            style={
              Object {
                "height": "100%",
                "paddingLeft": 20,
                "width": "70%",
              }
            }
          >
            <Text>
              testTitle1
            </Text>
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignSelf": "flex-start",
                "backgroundColor": "gray",
                "borderColor": "#0000001A",
                "borderRadius": 4,
                "borderWidth": 0,
                "marginRight": 20,
                "opacity": 1,
              }
            }
          >
            <Image
              source={
                Object {
                  "testUri": "../../react-native-ui-kitten/src/assets/img/choice/empty.png",
                }
              }
              style={
                Array [
                  Array [
                    Object {
                      "width": 20,
                    },
                    Object {
                      "height": 20,
                    },
                    Object {
                      "justifyContent": "center",
                    },
                    Object {
                      "alignItems": "center",
                    },
                    Object {
                      "margin": 4,
                    },
                  ],
                  null,
                ]
              }
            />
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "borderBottomColor": "gray",
              "borderBottomWidth": 1,
              "flexDirection": "row",
              "height": 50,
              "justifyContent": "space-between",
              "marginVertical": 5,
            }
          }
        >
          <View
            style={
              Object {
                "height": "100%",
                "paddingLeft": 20,
                "width": "70%",
              }
            }
          >
            <Text>
              testTitle2
            </Text>
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignSelf": "flex-start",
                "backgroundColor": "gray",
                "borderColor": "#0000001A",
                "borderRadius": 4,
                "borderWidth": 0,
                "marginRight": 20,
                "opacity": 1,
              }
            }
          >
            <Image
              source={
                Object {
                  "testUri": "../../react-native-ui-kitten/src/assets/img/choice/empty.png",
                }
              }
              style={
                Array [
                  Array [
                    Object {
                      "width": 20,
                    },
                    Object {
                      "height": 20,
                    },
                    Object {
                      "justifyContent": "center",
                    },
                    Object {
                      "alignItems": "center",
                    },
                    Object {
                      "margin": 4,
                    },
                  ],
                  null,
                ]
              }
            />
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
